account-management-service/src/main/java/org/bf2/srs/fleetmanager/spi/ams/impl/AccountManagementServiceWrapper.java

```diff
@@ -0,0 +1,74 @@
+package org.bf2.srs.fleetmanager.spi.ams.impl;
+
+import io.micrometer.core.annotation.Timed;
+import io.quarkus.arc.profile.UnlessBuildProfile;
+import org.bf2.srs.fleetmanager.common.metrics.Constants;
+import org.bf2.srs.fleetmanager.common.operation.auditing.Audited;
+import org.bf2.srs.fleetmanager.common.operation.faulttolerance.FaultToleranceConstants;
+import org.bf2.srs.fleetmanager.common.operation.faulttolerance.RetryUnwrap;
+import org.bf2.srs.fleetmanager.common.operation.faulttolerance.RetryWrap;
+import org.bf2.srs.fleetmanager.common.operation.faulttolerance.RetryWrapperException;
+import org.bf2.srs.fleetmanager.spi.ams.AccountManagementService;
+import org.bf2.srs.fleetmanager.spi.ams.AccountManagementServiceException;
+import org.bf2.srs.fleetmanager.spi.ams.ResourceLimitReachedException;
+import org.bf2.srs.fleetmanager.spi.ams.SubscriptionNotFoundServiceException;
+import org.bf2.srs.fleetmanager.spi.ams.TermsRequiredException;
+import org.bf2.srs.fleetmanager.spi.common.model.AccountInfo;
+import org.bf2.srs.fleetmanager.spi.common.model.ResourceType;
+import org.eclipse.microprofile.faulttolerance.Retry;
+import org.eclipse.microprofile.faulttolerance.Timeout;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import static java.util.Objects.requireNonNull;
+import static org.bf2.srs.fleetmanager.common.operation.auditing.AuditingConstants.KEY_AMS_SUBSCRIPTION_ID;
+
+@ApplicationScoped
+@UnlessBuildProfile("test")
+public class AccountManagementServiceWrapper implements AccountManagementService {
+
+    @Inject
+    AccountManagementServiceProducer producer;
+
+    private AccountManagementService delegate;
```

Comment:
Why not let the producer do its thing and then inject here the produced bean?

Reply:
Because the wrapper implements the same interface as the produced bean, would lead to cycle.

Reply:
Or I would have to add some workaround to avoid it. This seemed simpler.